dbt/adapters/snowflake/relation.py

```diff
@@ -33,3 +51,48 @@ def is_dynamic_table(self) -> bool:
     @classproperty
     def DynamicTable(cls) -> str:
         return str(SnowflakeRelationType.DynamicTable)
+
+    @classproperty
+    def get_relation_type(cls) -> Type[SnowflakeRelationType]:
+        return SnowflakeRelationType
+
+    @classmethod
+    def from_runtime_config(cls, runtime_config: RuntimeConfigObject) -> RelationConfigBase:
+        model_node: ModelNode = runtime_config.model
+        relation_type = SnowflakeRelationType(model_node.config.materialized)
+
+        if relation_config := cls.relation_configs.get(relation_type):
+            return relation_config.from_model_node(model_node)
+
+        raise DbtRuntimeError(
+            f"from_runtime_config() is not supported for the provided relation type: {relation_type}"
+        )
+
+    @classmethod
+    def dynamic_table_config_changeset(
```

Comment:
So this function is building the changeset between the new and existing version of the table? I find the name doesn't capture these semantics, and it would help self-document

Reply:
I went around the block on naming this and I agree that it's not super intuitive. I'll see if I can come up with something.

For context, here was my thought process leading up to this name. At this point in the process, we know that there's an existing database object and an object in the config that points to that database object, which may or may not have changes. Both things are really the same thing, the same dynamic table. We're merely taking another snapshot and then determining if the state of the object (minus data) in the config matches the already existing object in the database. Since they both represent the same dynamic table at two points in time, we're effectively determining the changeset between those two points in time on the same dynamic table; hence `dynamic_table_config_changeset()`. This makes me wonder if maybe the method name isn't that bad, but the arguments should be updated, as you point out below.

Reply:
I'm fine with more specific args and doc strings!